server/models/contests/addModerator.js

```diff
@@ -13,30 +13,17 @@ function addModerator({ params, body, username }) {
     const { contest_id: contestId } = params
     const { moderator: moderatorUsername } = body
     pool.query(
-      `SELECT * FROM contests_moderators WHERE contest_id=? AND moderator=?`,
-      [contestId, username],
-      (error, results) => {
-        if (error) {
+      `INSERT INTO contests_moderators (contest_id, moderator) SELECT contest_id, ? FROM contests_moderators WHERE contest_id=? AND moderator=?`,
+      [moderatorUsername, contestId, username],
+      (error, res) => {
+        if (error || res === undefined) {
+          const { code } = error
+          if (code === 'ER_DUP_ENTRY') {
+            return reject('The user is already a moderator')
+          }
```

Comment:
Similar as above. Appropriate error should be returned when user doesn't have moderator access